tpu-provisioner/test/e2e/test/util/cleanup.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+// See: https://github.com/golang/go/issues/41891
+
+import (
+	"os"
+	"os/signal"
+	"sync"
+	"testing"
+)
+
+// cleanup allows you to define a cleanup function that will be executed
+// if your test is interrupted.
+
+func init() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	go cleanupOnInterrupt(c)
+}
+
+var cf struct {
+	o sync.Once
+	m sync.RWMutex
+	f []func()
+}
```

Comment:
The variable names for this struct and its fields are unclear to me, can we make them more descriptive?

I'm guessing:

"cf" = ?
 "o" = "once"
 "m" = "mutex"
 "f" = "functions"